eos/form-factors/analytic-d-to-pi.cc

```diff
@@ -0,0 +1,2177 @@
+/* vim: set sw=4 sts=4 et foldmethod=syntax : */
+
+/*
+ * Copyright (c) 2014, 2015, 2020 Danny van Dyk
+ * Copyright (c) 2019, 2020 Domagoj Leljak
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/form-factors/analytic-d-to-pi.hh>
+#include <eos/form-factors/pi-lcdas.hh>
+#include <eos/maths/derivative.hh>
+#include <eos/utils/exception.hh>
+#include <eos/maths/integrate.hh>
+#include <eos/utils/log.hh>
+#include <eos/models/model.hh>
+#include <eos/utils/options-impl.hh>
+#include <eos/maths/polylog.hh>
+#include <eos/maths/power-of.hh>
+#include <eos/utils/private_implementation_pattern-impl.hh>
+#include <eos/utils/qcd.hh>
+
+#include <functional>
+#include <limits>
+
+#include <gsl/gsl_sf_gamma.h>
+
+namespace eos
+{
+    template <>
+    struct Implementation<AnalyticFormFactorDToPiDKMMO2008>
+    {
+        std::shared_ptr<Model> model;
+
+        // hadronic parameters
+        UsedParameter MD;
+        UsedParameter mpi;
+        UsedParameter fpi;
+
+        // Borel parameters, thresholds and renormalization scale
+        SwitchOption opt_rescale_borel;
+        std::function<double (const double &)> rescale_factor_p;
+        std::function<double (const double &)> rescale_factor_0;
+        std::function<double (const double &)> rescale_factor_T;
+        UsedParameter M2;
+        UsedParameter Mprime2;
+        UsedParameter _s0_plus, _s0_plus_p, _s0_plus_pp;
+        UsedParameter _s0_zero, _s0_zero_p, _s0_zero_pp;
+        UsedParameter _s0_T,    _s0_T_p,    _s0_T_pp;
+        UsedParameter sprime0B;
+        UsedParameter mu;
+
+        // Parameter for the estimation of NNLO corrections
+        UsedParameter zeta_nnlo;
+
+        // QCD parameters
+        UsedParameter m02;
+        UsedParameter cond_GG;
+        UsedParameter r_vac;
+
+        PionLCDAs pi;
+
+        GSL::QAGS::Config config;
+
+        static const std::vector<OptionSpecification> options;
+
+        Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+            model(Model::make("SM", p, o)),
+            MD(p["mass::D_d"], u), //B_d -> D_d
+            mpi(p["mass::pi^+"], u),
+            fpi(p["decay-constant::pi"], u),
+            opt_rescale_borel(o, "rescale-borel", { "1", "0" }, "1"),
+            M2(p["B->pi::M^2@DKMMO2008"], u),
```

Comment:
The names of the parameters need to be changed now: ``"D->pi::M^2@DKMMO2008"``, and also below!

Reply:
I changed also references to KMMO2009